pkg/virt-controller/watch/vm/vm_test.go

```diff
@@ -5507,6 +5524,107 @@ var _ = Describe("VirtualMachine", func() {
 					Expect(vm.Spec.Preference.RevisionName).To(Equal(expectedPreferenceRevision.Name))
 				})
 			})
+
+			Context("InstancetypeReferencePolicy", func() {
+				BeforeEach(func() {
+					vm.Spec.Instancetype = &v1.InstancetypeMatcher{
+						Name: instancetypeObj.Name,
+						Kind: instancetypeapi.SingularResourceName,
+					}
+					vm.Spec.Preference = &v1.PreferenceMatcher{
+						Name: preference.Name,
+						Kind: instancetypeapi.SingularPreferenceResourceName,
+					}
+				})
+
+				DescribeTable("should not expand and update VM ", func(kv *v1.KubeVirt, updateVMFunc func()) {
+					testutils.UpdateFakeKubeVirtClusterConfig(kvStore, kv)
+					updateVMFunc()
+
+					vm, err := virtFakeClient.KubevirtV1().VirtualMachines(vm.Namespace).Create(
+						context.TODO(), vm, metav1.CreateOptions{})
+					Expect(err).To(Succeed())
+
+					addVirtualMachine(vm)
+					sanityExecute(vm)
+
+					vm, err = virtFakeClient.KubevirtV1().VirtualMachines(vm.Namespace).Get(
+						context.TODO(), vm.Name, metav1.GetOptions{})
+					Expect(err).To(Succeed())
+					Expect(vm.Spec.Instancetype).ToNot(BeNil())
+					Expect(vm.Spec.Instancetype.RevisionName).ToNot(BeEmpty())
+					Expect(vm.Spec.Preference).ToNot(BeNil())
+					Expect(vm.Spec.Preference.RevisionName).ToNot(BeEmpty())
+				},
+					Entry("with default referencePolicy", &v1.KubeVirt{
+						Spec: v1.KubeVirtSpec{Configuration: v1.KubeVirtConfiguration{}}}, func() {}),
+					Entry("with referencePolicy reference", &v1.KubeVirt{
+						Spec: v1.KubeVirtSpec{
+							Configuration: v1.KubeVirtConfiguration{
+								Instancetype: &v1.InstancetypeConfiguration{
+									ReferencePolicy: pointer.P(v1.Reference),
+								},
+							},
+						},
+					}, func() {}),
+					Entry("with referencePolicy expand and revisionNames already captured",
+						&v1.KubeVirt{
+							Spec: v1.KubeVirtSpec{
+								Configuration: v1.KubeVirtConfiguration{
+									Instancetype: &v1.InstancetypeConfiguration{
+										ReferencePolicy: pointer.P(v1.Expand),
+									},
+								},
+							},
+						},
+						func() {
+							instancetypeRevision, err := instancetype.CreateControllerRevision(vm, instancetypeObj)
+							Expect(err).ToNot(HaveOccurred())
+
+							_, err = virtClient.AppsV1().ControllerRevisions(vm.Namespace).Create(
```

Comment:
Do these actually need to exist in unit tests?

Reply:
Discussed and yes as the upgrade logic runs in the same reconcile loop.